src/ape_pm/_cli.py

```diff
@@ -321,10 +314,35 @@ def compile(cli_ctx, name, version, force, config_override):
         if config_override:
             dependency.api.config_override = config_override
 
-        try:
-            dependency.compile(use_cache=not force)
-        except Exception as err:
-            cli_ctx.logger.error(str(err))
-            continue
-        else:
+        _compile_dependency(cli_ctx, dependency, force)
+
+
+def _compile_dependency(cli_ctx, dependency: Dependency, force: bool):
+    try:
+        result = dependency.compile(use_cache=not force)
+    except Exception as err:
+        cli_ctx.logger.error(str(err))
+    else:
+        if result:
```

Comment:
- [ ] Move this check to `dependency.compile()` logic so you get the warning in regular python code as well (?)